src/hooks/spaces/useSpaceBySlug.ts

```diff
@@ -0,0 +1,63 @@
+import { useState } from "react";
+import { useFirestore, useFirestoreCollectionData } from "reactfire";
+import Bugsnag from "@bugsnag/js";
+
+import { AnyVenue } from "types/venues";
+
+import { withIdConverter } from "utils/converters";
+import { WithId } from "utils/id";
+
+type UseSpaceBySlugResult = {
+  space?: WithId<AnyVenue>;
+  isLoaded: boolean;
+};
+
+/**
+ * Hook which will return the space when the slug is provided.
+ * The intention is to be used on the client side, when the space slug is provided in the url.
+ * @param spaceSlug
+ * @returns
+ */
+export const useSpaceBySlug: (spaceSlug?: string) => UseSpaceBySlugResult = (
+  spaceSlug
+) => {
+  const [error, setError] = useState<string>();
```

Comment:
This will make the error a singleton shared by all components that try to get a hold of a world. Not necessary to keep it as a state once returned and might cause shared memory problems if it is kept.
```suggestion
```

Reply:
Very true. I can't believe I missed that.

Reply:
Fixed in: https://github.com/sparkletown/sparkle/pull/2548/commits/9cc14823a26b81030e32d035218556fed2c2c758